sinks/gcm/driver.go

```diff
@@ -185,6 +203,39 @@ func (self *gcmSink) StoreTimeseries(input []sink_api.Timeseries) error {
 		})
 	}
 
+	// Only send one metric of each type per request.
+	var lastErr error
+	for len(metrics) != 0 {
+		var request metricWriteRequest
+		for name, values := range metrics {
+			// Remove metrics with no more values.
+			if len(values) == 0 {
+				delete(metrics, name)
+				continue
+			}
+
+			m := values[0]
+			metrics[name] = values[1:]
+			request.Timeseries = append(request.Timeseries, m)
+		}
+
+		err := self.pushMetrics(&request)
+		if err != nil {
+			lastErr = err
+		}
+	}
+
+	return lastErr
+}
+
+func (self *gcmSink) pushMetrics(request *metricWriteRequest) error {
+	if len(request.Timeseries) == 0 {
+		return nil
+	}
+	if len(request.Timeseries) > maxTimeseriesPerRequest {
+		return fmt.Errorf("unable to write more than %d metrics at once and %d were provided", maxTimeseriesPerRequest, len(request.Timeseries))
```

Comment:
True. Once we start adding custom metrics, wouldn't we hit this condition?

On Fri, Mar 20, 2015 at 3:59 PM, Victor Marmol <notifications@github.com>
wrote:

> In sinks/gcm/driver.go
> <https://github.com/GoogleCloudPlatform/heapster/pull/176#discussion_r26883098>
> :
>
> > +
> > +		err := self.pushMetrics(&request)
> > +		if err != nil {
> > +			lastErr = err
> > +		}
> > +	}
> > +
> > +	return lastErr
> > +}
> > +
> > +func (self *gcmSink) pushMetrics(request *metricWriteRequest) error {
> > +	if len(request.Timeseries) == 0 {
> > +		return nil
> > +	}
> > +	if len(request.Timeseries) > maxTimeseriesPerRequest {
> > +		return fmt.Errorf("unable to write more than %d metrics at once and %d were provided", maxTimeseriesPerRequest, len(request.Timeseries))
>
> This is just a sanity check, by now we should have done this.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/heapster/pull/176/files#r26883098>
> .
>